### PR TITLE
SPM: Remove SPM from NCS

### PR DESCRIPTION
**Pull request by joerchan**
Remove SPM and secure services
Remove documentation of SPM and secure services
Remove SPM noup patches in sdk-zephyr
Remove SPM support in the entropy driver.

Document that SPM has been removed from NCS.

### CODE REVIEW

**Review by hakonfam — CHANGES_REQUESTED** [review]
We should remove from all partition manager configs as well 
image

There is also this:
image

and this 
image

**Review by plskeggs — APPROVED** [review]

**Review by hakonfam — APPROVED** [review]

**Review by simensrostad — APPROVED** [review]

**Review by glarsennordic — APPROVED** [review]
LGTM

**Review by gregersrygg — APPROVED** [review]

**Review by sigvartmh — APPROVED** [review]

**Review by mia-ko — CHANGES_REQUESTED** [review]
What about samples\bootloader\pm.yml?

File: doc/nrf/releases/release-notes-1.0.0.rst [hunk 1]
```diff
@@ -230,12 +230,12 @@ Documentation
 
   * nRF9160:
 
-    * :ref:`lib_spm`
+    * Secure Partition Manager (SPM)
```
[the remark this answers is not in the document]
Comment (joerchan) [reply doc/nrf/releases/release-notes-1.0.0.rst:233]: This was a search replace, `lib_spm` would have been replaced with this text.

File: west.yml [hunk 2]
```diff
@@ -52,7 +52,7 @@ manifest:
     # https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/modules.html
     - name: zephyr
       repo-path: sdk-zephyr
-      revision: 707ff27ecaad44b00cc015bcf0c46db700a4267f
+      revision: pull/1005/head
```
Comment (rlubos) [inline west.yml:55]: Please update SHA: `8fb465f43f6a90ac4c4ce89265047e04f7d5ee5b`

File: doc/nrf/known_issues.rst [hunk 3]
```diff
@@ -1853,20 +1853,20 @@ Secure Partition Manager (SPM)
 ==============================
 
 .. note::
-    As of the |NCS| v2.1.0, Secure Partition Manager (SPM) is deprecated and replaced by :ref:`Trusted Firmware-M (TF-M) <ug_tfm>`.
+    The Secure Partition Manager (SPM) was deprecated in |NCS| 2.1.0 and removed in |NCS| v2.3.0, it is replaced by :ref:`Trusted Firmware-M (TF-M) <ug_tfm>`.
```
Comment (mia-ko) [inline doc/nrf/known_issues.rst:1856]: Sorry, I missed this one in the first round. It's better to split it in two sentences.
```suggestion
    The Secure Partition Manager (SPM) was deprecated in |NCS| 2.1.0 and removed in |NCS| v2.3.0. It is replaced by :ref:`Trusted Firmware-M (TF-M) <ug_tfm>`.
```

**Review by mia-ko — CHANGES_REQUESTED** [review]
Just one tiny thing I missed in the first round...

And one more thing. I found some SPM related stuff in sample_boardnames.txt (in doc\nrf\includes\boardname_tables), see line 268-281... I'm thinking we could remove this as well :)

**Review by mia-ko — APPROVED** [review]

**Pull request merged**